src/XrmContext/Resources/XrmExtensions.cs

```diff
@@ -63,6 +63,33 @@ protected void SetOptionSetValue<T>(string attributeName, T value)
             }
         }
 
+        protected IEnumerable<T> GetOptionSetCollectionValue<T>(string attributeName) where T : struct, IComparable, IConvertible, IFormattable
+        {
+            var optionSetCollection = GetAttributeValue<OptionSetValueCollection>(attributeName);
+            if (optionSetCollection != null && optionSetCollection.Count != 0)
+            {
+                return optionSetCollection
+                    .Select(osv => (T)Enum.ToObject(typeof(T), osv.Value))
+                    .ToArray();
+            }
+            return null;
+        }
+
+        protected void SetOptionSetCollectionValue<T>(string attributeName, IEnumerable<T> value)
```

Comment:
From a purely UX perspective: 
It would be nice to also have a separate setter method that takes a variable amount of values (using [params](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/params)). That way developers are able to just pass a single or multiple singular-values, and won't have to always create an enumerable each time they want to set values on an OptionSetCollection.